pkg/network/ovn_kubernetes.go

```diff
@@ -342,3 +378,38 @@ func listenDualStack(masterIP string) string {
 		return ""
 	}
 }
+
+// shouldUpdateOvnKMaster determines if we're ready to roll out
+// upgrades to the master daemonset.
+//
+// We shouldn't upgrade ovn-k master until all the nodes are upgraded, because of
+// compatability questions
+func shouldUpdateMaster(existingNode, existingMaster *appsv1.DaemonSet) bool {
+	releaseVersion := os.Getenv("RELEASE_VERSION")
+	// Fresh cluster - full steam ahead!
+	if existingNode == nil || existingMaster == nil {
+		return true
+	}
+
+	nodeVersion := existingNode.GetAnnotations()["release.openshift.io/version"]
+
+	if nodeVersion != releaseVersion {
+		// TODO: expose this directly as status. Need to refactor status-manager
+		klog.V(3).Infof("Not upgrading ovn-kube master yet, ovn-kube node DaemonSet still at version %s, need %s",
+			nodeVersion, releaseVersion)
+		return false
+	}
+
+	status := existingNode.Status
+	progressing := (status.UpdatedNumberScheduled < status.DesiredNumberScheduled ||
+		status.NumberUnavailable > 0 ||
+		status.NumberAvailable == 0 ||
+		existingNode.Generation > status.ObservedGeneration)
```

Comment:
This seems very magical and really needs at least a comment

Reply:
Agreed.